packages/x-components/src/x-modules/scroll/components/main-scroll.vue

```diff
@@ -0,0 +1,403 @@
+<script lang="ts">
+  import { Component, Prop, Watch } from 'vue-property-decorator';
+  import { xComponentMixin } from '../../../components/x-component.mixin';
+  import { State } from '../../../components/decorators/store.decorators';
+  import { XOn } from '../../../components/decorators/bus.decorators';
+  import { XProvide } from '../../../components/decorators/injection.decorators';
+  import { NoElement } from '../../../components/no-element';
+  import { scrollXModule } from '../x-module';
+  import { FirstVisibleItemObserverKey } from './scroll.const';
+  import { ScrollVisibilityObserver } from './scroll.types';
+
+  /**
+   * Extends the scroll making it able to sync the first visible element, and allowing
+   * the children position to be restored.
+   *
+   * Each child element that wants to have this support must be wrapped in a {@link MainScrollItem}
+   * component.
+   *
+   * @public
+   */
+  @Component({
+    mixins: [xComponentMixin(scrollXModule)]
+  })
+  /* eslint-disable @typescript-eslint/unbound-method */
+  export default class MainScroll extends NoElement {
+    /**
+     * The reference to the HTML node that has the scroll.
+     *
+     * @public
+     */
+    public $el!: HTMLElement;
```

Comment:
Is this attribute really needed here? `NoElement` is a Vue instance with `$el` defined

Reply:
Vue `$el` has type `Element`, I'm narrowing that type to a more concrete one to avoid the `as HTMLElement` casts everywhere.